modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceAdapter.java

```diff
@@ -123,10 +124,11 @@ public SourceAdapter(ExtensionModel extensionModel, SourceModel sourceModel,
     this.extensionModel = extensionModel;
     this.sourceModel = sourceModel;
     this.source = source;
+    this.extensionMessageSource = extensionMessageSource;
     this.cursorProviderFactory = cursorProviderFactory;
     this.configurationInstance = configurationInstance;
     this.sourceCallbackFactory = sourceCallbackFactory;
-    this.componentLocation = source.getLocation();
+    this.componentLocation = extensionMessageSource.getLocation();
```

Comment:
this doesn't make sense to me... You generate a cyclic dependency in which the ExtensionMessageSource knows of the SourceAdapter and vice-versa, while signficant effort has been made to keep those decoupled, all for the mere sake of doing `extnesionMessageSource.getLocation()`. Just pass the location over. That keeps things decoupled and more testeable